Source/Core/GoogleEarthEnterpriseTerrainProvider.js

```diff
@@ -0,0 +1,548 @@
+        HeightmapTerrainData,
+        loadArrayBuffer,
+        CesiumMath,
+        Rectangle,
+        RuntimeError,
+        TerrainProvider,
+        throttleRequestByServer,
+        TileProviderError,
+        when) {
+    'use strict';
+
+    var TerrainState = {
+        UNKNOWN : 0,
+        NONE : 1,
+        SELF : 2,
+        PARENT : 3
+    };
+
+    /**
+     * Provides tiled terrain using the Google Earth Enterprise REST API.
+     *
+     * @alias GoogleEarthEnterpriseTerrainProvider
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {String} options.url The url of the Google Earth Enterprise server hosting the imagery.
+     * @param {GoogleEarthEnterpriseMetadata} options.metadata A metadata object that can be used to share metadata requests with a GoogleEarthEnterpriseImageryProvider.
+     * @param {Proxy} [options.proxy] A proxy to use for requests. This object is
+     *        expected to have a getURL function which returns the proxied URL, if needed.
+     * @param {Ellipsoid} [options.ellipsoid] The ellipsoid.  If not specified, the WGS84 ellipsoid is used.
+     * @param {Credit|String} [options.credit] A credit for the data source, which is displayed on the canvas.
+     *
+     * @see GoogleEarthEnterpriseImageryProvider
+     * @see CesiumTerrainProvider
+     *
+     * @example
+     * var geeMetadata = new GoogleEarthEnterpriseMetadata('http://www.earthenterprise.org/3d');
+     * var gee = new Cesium.GoogleEarthEnterpriseTerrainProvider({
+     *     metadata : geeMetadata
+     * });
+     *
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     */
+    function GoogleEarthEnterpriseTerrainProvider(options) {
+        options = defaultValue(options, {});
+
+        //>>includeStart('debug', pragmas.debug);
+        if (!(defined(options.url) || defined(options.metadata))) {
+            throw new DeveloperError('options.url or options.metadata is required.');
+        }
+        //>>includeEnd('debug');
+
+        if (defined(options.metadata)) {
+            this._metadata = options.metadata;
+        } else {
+            this._metadata = new GoogleEarthEnterpriseMetadata({
+                url : options.url,
+                proxy : options.proxy
+            });
+        }
+        this._proxy = defaultValue(options.proxy, this._metadata.proxy);
+
+        this._tilingScheme = new GeographicTilingScheme({
+            numberOfLevelZeroTilesX : 2,
+            numberOfLevelZeroTilesY : 2,
+            rectangle : new Rectangle(-CesiumMath.PI, -CesiumMath.PI, CesiumMath.PI, CesiumMath.PI),
+            ellipsoid : options.ellipsoid
+        });
+
+        var credit = options.credit;
+        if (typeof credit === 'string') {
+            credit = new Credit(credit);
+        }
+        this._credit = credit;
+
+        // Pulled from Google's documentation
+        this._levelZeroMaximumGeometricError = 40075.16;
```

Comment:
For my understanding, where does the per-tile geometric error come from?  Is it stored as part of the tile or estimated from the level?